spec/services/commitments/calculate_amount_service_spec.rb

```diff
@@ -0,0 +1,261 @@
+
+            expect(result.commitment_amount_cents).to eq(commitment.amount_cents)
+          end
+        end
+      end
+    end
+
+    context 'when plan has quarterly interval' do
+      let(:amount_cents) { 40_000 }
+      let(:interval) { :quarterly }
+      let(:from_datetime) { DateTime.parse('2024-01-15T00:00:00') }
+      let(:to_datetime) { DateTime.parse('2024-03-31T23:59:59') }
+
+      context 'when subscription is calendar' do
+        let(:billing_time) { :calendar }
+
+        context 'when there is no commitment' do
+          let(:commitment) { nil }
+
+          it 'returns result' do
+            result = apply_service.call
+
+            expect(result.commitment_amount_cents).to eq(0)
+          end
+        end
+
+        context 'when a commitment exists for a plan' do
+          let(:commitment) { create(:commitment, plan:, amount_cents:) }
+
+          before { commitment }
+
+          it { is_expected.to delegate_method(:subscription).to(:invoice_subscription) }
+
+          it 'returns result' do
+            result = apply_service.call
+
+            expect(result.commitment_amount_cents).to eq(33_846)
+          end
+        end
+      end
+
+      context 'when subscription is anniversary' do
+        let(:billing_time) { :anniversary }
+
+        context 'when there is no commitment' do
+          let(:commitment) { nil }
+
+          it 'returns result' do
+            result = apply_service.call
+
+            expect(result.commitment_amount_cents).to eq(0)
+          end
+        end
+
+        context 'when a commitment exists for a plan' do
+          let(:commitment) { create(:commitment, plan:, amount_cents:) }
+
+          it { is_expected.to delegate_method(:subscription).to(:invoice_subscription) }
+
+          it 'returns result' do
+            result = apply_service.call
+
+            expect(result.commitment_amount_cents).to eq(commitment.amount_cents)
+          end
+        end
+      end
+    end
+
+    context 'when plan has yearly interval' do
+      let(:amount_cents) { 200_000 }
+      let(:interval) { :yearly }
+      let(:from_datetime) { DateTime.parse('2024-01-15T00:00:00') }
+      let(:to_datetime) { DateTime.parse('2024-12-31T23:59:59') }
+
+      context 'when subscription is calendar' do
+        let(:billing_time) { :calendar }
+
+        context 'when there is no commitment' do
+          let(:commitment) { nil }
+
+          it 'returns result' do
+            result = apply_service.call
+
+            expect(result.commitment_amount_cents).to eq(0)
+          end
+        end
+
+        context 'when a commitment exists for a plan' do
+          let(:commitment) { create(:commitment, plan:, amount_cents:) }
+
+          before { commitment }
+
+          it { is_expected.to delegate_method(:subscription).to(:invoice_subscription) }
+
+          it 'returns result' do
+            result = apply_service.call
+
+            expect(result.commitment_amount_cents).to eq(192350)
```

Comment:
Style/NumericLiterals: Use underscores(_) as thousands separator and separate every 3 digits with them. (https://rubystyle.guide#underscores-in-numerics)